### PR TITLE
Java: Casting

### LINKED ISSUE #1533: Java 3.6: Casting
<hr>

## Description 📜

Add audio, documentation, and video regarding the **Casting**.

<hr>

## Domain of Contribution 📊

- [x] Java

<hr>

## Location of File to be added

The files should be added inside the `Casting` folder which is inside the `Primitive Data Types` folder present in the  `Java` main folder.  The documentation file can be in the `.md`/`.py`/`.ipynb` format. Audio should be in `.mp3` format and to be submitted through a Google drive link. Video can be in any format and should be submitted through a Google Drive link.

 **Note:**
 - You are required to comment with `\assign CONTENT_TYPE`, so that other content types can be created at the same time.
 - Failing to comment in the above manner will lead to the removal of assignment from the issue even if the bot assigns.
 - You are required to do a PR within 7 days otherwise you will be unassigned and others will be assigned.
 - Issues will be assigned on a first come first serve basis.

### PR DESCRIPTION
**Pull request by ketkikaushik**
<hr>

## Description 📜

This documentation contains detailed explanation of  Casting in Java and its types with examples.
Fixes #1533 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Java_Domain

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by jaigora24** [comment]
https://github.com/girlscript/winter-of-contributing/pull/3811 Review my PR

**Comment by AkMo3** [comment]
@jaigora24 It is a PR, not an issue that can be assigned.

File: Java/Primitive Data Type/Casting/Type_Casting.ipynb [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "P7G7MtOqVjD1"
+      },
+      "source": [
+        "In the above example, we are assigning 0 (an integer value) to 'a' (an integer variable) and that works fine.\n",
```
Comment (sakshamlamba109) [inline Java/Primitive Data Type/Casting/Type_Casting.ipynb:45]: In the example the contributor has int a=`7`, but in this line, the contributor has written "we are assigning  `0`"

File: Java/Primitive Data Type/Casting/Type_Casting.ipynb [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "MMUnFI1Zd39E"
+      },
+      "source": [
+        "# FACTS:\n",
+        "\n",
+        "\n",
+        "*   Java will *upcast* a variable *automatically*.\n",
+        "*   We must *downcast* a variable *manuall*.\n",
```
Comment (sakshamlamba109) [inline Java/Primitive Data Type/Casting/Type_Casting.ipynb:79]: spelling error

File: Java/Primitive Data Type/Casting/Type_Casting.ipynb [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "VWlEVwQibFld"
+      },
+      "source": [
+        "# IMPLICIT TYPE CASTING\n",
+        "In Implicit type casting/conversion the data type gets automatically converted to another data type by the compiler. It is also known as Widening type casting as it converts lower data type into higher one and so there is no loss of data in this.\n",
```
Comment (sakshamlamba109) [inline Java/Primitive Data Type/Casting/Type_Casting.ipynb:106]: please mention lower priority data type instead of lower data type

**Review by sakshamlamba109 — CHANGES_REQUESTED** [review]
@garvitraj I've mentioned the changes

**Review by sakshamlamba109 — CHANGES_REQUESTED** [review]
@AkMo3 please look into it

File: Java/Primitive Data Type/Casting/Type_Casting.ipynb (hunk 2 above)
Comment (AkMo3) [inline Java/Primitive Data Type/Casting/Type_Casting.ipynb:79]: `manuall` -> `manual`

File: Java/Primitive Data Type/Casting/Type_Casting.ipynb [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        "\n",
+        "There are two types of casting:\n",
+        "\n",
+        "1) Implicict type casting\n",
```
Comment (AkMo3) [inline Java/Primitive Data Type/Casting/Type_Casting.ipynb:94]: `Implicict` -> `Implicit`

File: Java/Primitive Data Type/Casting/Type_Casting.ipynb [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Hsg4JTkmmWB5"
+      },
+      "source": [
+        "# EXPLICIT TYPE CASTING\n",
+        "In Explicit casting, the coversion does not happen automatically. It should be coverted by a programmer when converting from a wider to a narrower data type. It is also known as Narrowing type casting. If we do not perform casting, it will cause compile-time error.\n",
```
Comment (AkMo3) [inline Java/Primitive Data Type/Casting/Type_Casting.ipynb:192]: `coversion` -> `conversion`
`coverted` -> `converted`

File: Java/Primitive Data Type/Casting/Type_Casting.ipynb [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+        "Now, \"4.5 + 3.0\" will be calculated as \"7.5\" and stored in **d1**."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "jc8TkBVs3wD7"
+      },
+      "source": [
+        "Referances: [javatpoint](https://www.javatpoint.com/type-casting-in-java)\n",
```
Comment (AkMo3) [inline Java/Primitive Data Type/Casting/Type_Casting.ipynb:289]: `Referances` -> `References`

**Review by AkMo3 — CHANGES_REQUESTED** [review]

**Comment by AkMo3** [comment]
@ketkikaushik Please do all the changes suggested by Saksham and Me

**Comment by ketkikaushik** [comment]
> @ketkikaushik Please do all the changes suggested by Saksham and Me

ok I'll work on that.

**Comment by ketkikaushik** [comment]
@AkMo3 I've made all the changes that you requested. Please review it.

**Review by AkMo3 — APPROVED** [review]

**Pull request merged**